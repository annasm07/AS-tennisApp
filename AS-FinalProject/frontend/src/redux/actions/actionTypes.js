export default {
  LOG_IN: 'LOG_IN',
  LOG_IN_ERROR: 'LOG_IN_ERROR',
  SIGN_UP: 'LOG_IN',
  SIGN_UP_ERROR: 'LOG_IN_ERROR',
  GET_PLAYER: 'GET_PLAYER',
  GET_PLAYER_ERROR: 'GET_PLAYER_ERROR',
  GET_PLAYERS: 'GET_PLAYERS',
  GET_PLAYERS_ERROR: 'GET_PLAYERS_ERROR',
  CREATE_MATCH: 'CREATE_MATCH',
  CREATE_MATCH_ERROR: 'CREATE_MATCH_ERROR',
  UPDATE_POINTS: 'UPDATE_POINTS',
  UPDATE_POINTS_ERROR: 'UPDATE_POINTS_ERROR',
  UPDATE_GAMES: 'UPDATE_GAMES',
  UPDATE_GAMES_ERROR: 'UPDATE_GAMES_ERROR',
  END_GAME: 'END_GAME',
  END_GAME_ERROR: 'END_GAME_ERROR',
  END_SET: 'END_SET',
  END_SET_ERROR: 'END_SET_ERROR',
  END_MATCH: 'END_MATCH',
  END_MATCH_ERROR: 'END_MATCH_ERROR',
};
