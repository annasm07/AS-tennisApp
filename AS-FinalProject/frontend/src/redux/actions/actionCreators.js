import axios from 'axios';
import actionTypes from './actionTypes';

const LOG_IN_URL = 'http://localhost:1616/api/auth/login';
const SIGN_UP_URL = 'http://localhost:1616/api/auth/signup';
const GET_PLAYER_URL = 'http://localhost:1616/api/home';
const GET_ALL_PLAYERS = 'http://localhost:1616/api/players';
const CREATE_MATCH_URL = 'http://localhost:1616/api/match';

export function logIn(email, password) {
  return async dispatch => {
    try {
      const {data} = await axios.post(LOG_IN_URL, {email, password});
      dispatch({
        type: actionTypes.LOG_IN,
        user: data,
      });
    } catch (error) {
      console.log(error);
      dispatch({
        type: actionTypes.LOG_IN_ERROR,
      });
    }
  };
}

export function signUp({name, email, password, player, playerName}) {
  return async dispatch => {
    try {
      await axios.post(SIGN_UP_URL, {
        name,
        email,
        password,
        player,
        playerName,
      });
      dispatch({
        // type: actionTypes.SIGN_UP,
        // user: data.user,
      });
    } catch (error) {
      console.log(error);
      dispatch({
        type: actionTypes.SIGN_UP_ERROR,
      });
    }
  };
}

export function getPlayerInfo(token, playerId) {
  return async dispatch => {
    try {
      const {data} = await axios(`${GET_PLAYER_URL}/${playerId}`, {
        headers: {
          Authorization: `Bearer ${token}`,
        },
      });
      dispatch({
        type: actionTypes.GET_PLAYER,
        player: data,
      });
    } catch (error) {
      console.log(error);
      dispatch({
        type: actionTypes.GET_PLAYER_ERROR,
      });
    }
  };
}
export function getAllPlayers(token) {
  return async dispatch => {
    try {
      const {data} = await axios(GET_ALL_PLAYERS, {
        headers: {
          Authorization: `Bearer ${token}`,
        },
      });
      dispatch({
        type: actionTypes.GET_PLAYERS,
        players: data,
      });
    } catch (error) {
      console.log(error);
      dispatch({
        type: actionTypes.GET_PLAYERS_ERROR,
      });
    }
  };
}

export function newMatch(token, p1Name, p2Name) {
  return async dispatch => {
    try {
      const {data} = await axios.post(
        CREATE_MATCH_URL,
        {p1Name, p2Name},
        {
          headers: {
            Authorization: `Bearer ${token}`,
          },
        },
      );
      dispatch({
        type: actionTypes.CREATE_MATCH,
        currentMatch: data,
      });
    } catch (error) {
      console.log(error);
      dispatch({
        type: actionTypes.CREATE_MATCH_ERROR,
      });
    }
  };
}
