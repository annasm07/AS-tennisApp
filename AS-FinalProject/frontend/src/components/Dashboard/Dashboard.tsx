import React, {useEffect} from 'react';
import {useSelector, useDispatch} from 'react-redux';
import {
  SafeAreaView,
  View,
  Text,
  Image,
  StyleSheet,
  TouchableOpacity,
  ScrollView,
} from 'react-native';
import {getPlayerInfo} from '../../redux/actions/actionCreators';
import globalStyles from '../../theme/globalThemes';
import matchBoxStyles from '../../theme/matchBoxTheme';

function Dashboard({navigation}: any) {
  const player = useSelector((store: any) => store.player);
  const tokens = useSelector((store: any) => store.tokens);
  const user = useSelector((store: any) => store.user);
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getPlayerInfo(tokens[0], user.user.playerId));
  }, [dispatch, tokens, user]);

  function getDateString(dateString: any) {
    const dateToDisplay = new Date(dateString).toLocaleDateString();
    return dateToDisplay;
  }
  return (
    <SafeAreaView>
      <ScrollView>
        <View style={styles.playerBox}>
          <Image
            style={styles.playerImage}
            source={{
              uri: `${player.img}`,
            }}
          />
          <View style={styles.infoBox}>
            <Text style={styles.playerName}>{player.name}</Text>

            <View>
              {player.record && (
                <View style={styles.statsFullBar}>
                  <View
                    style={[
                      styles.statsWins,
                      {
                        width: `${
                          (player.record[0] /
                            (player.record[0] + player.record[1])) *
                          100
                        }%`,
                      },
                    ]}
                  />
                  <View
                    style={[
                      styles.statsLosses,
                      {
                        width: `${
                          (player.record[1] /
                            (player.record[0] + player.record[1])) *
                          100
                        }%`,
                      },
                    ]}
                  />
                </View>
              )}
              <View style={styles.recordInfo}>
                <Text style={globalStyles.grayText}>Wins / Losses</Text>
                {player.record && (
                  <Text style={globalStyles.grayText}>
                    {' '}
                    {player.record[0]} / {player.record[1]}
                  </Text>
                )}
              </View>
            </View>
          </View>
        </View>

        {player.playedMatches &&
          player.playedMatches.map((match: any) => (
            <TouchableOpacity
              testID="matchBox"
              key={match._id}
              onPress={() => {
                navigation.navigate('Stats', {
                  matchId: match._id,
                });
              }}>
              <View key={match._id}>
                <View style={matchBoxStyles.matchBox}>
                  <View style={matchBoxStyles.player1}>
                    <Text>{match.result[0].name}</Text>
                    <Text style={matchBoxStyles.result}>
                      {match.result[0].games}
                    </Text>
                  </View>
                  <View style={matchBoxStyles.player2}>
                    <Text>{match.result[1].name}</Text>
                    <Text style={matchBoxStyles.result}>
                      {match.result[1].games}
                    </Text>
                  </View>
                </View>
                <View style={matchBoxStyles.dateBox}>
                  <Text style={globalStyles.grayText}>
                    {getDateString(match.date)}
                  </Text>
                  <Text style={globalStyles.grayText}>Stats &gt;</Text>
                </View>
              </View>
            </TouchableOpacity>
          ))}
      </ScrollView>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  playerBox: {
    backgroundColor: '#E5E5E5',
    borderRadius: 10,
    padding: 10,
    margin: 15,
    flexDirection: 'row',
  },
  playerImage: {
    width: 80,
    height: 80,
    borderRadius: 100,
  },
  playerName: {
    marginTop: 10,
    fontSize: 20,
    paddingLeft: 20,
    textTransform: 'capitalize',
  },
  infoBox: {
    display: 'flex',
    flexDirection: 'column',
  },
  statsFullBar: {
    marginTop: 15,
    marginLeft: 20,
    width: 220,
    alignSelf: 'center',
    backgroundColor: 'pink',
    height: 14,
    display: 'flex',
    flexDirection: 'row',
  },
  statsWins: {
    width: '50%',
    backgroundColor: '#96F87E',
    height: 14,
  },
  statsLosses: {
    width: '50%',
    backgroundColor: '#F68A5B',
    height: 14,
  },
  recordInfo: {
    marginLeft: 20,
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
});

export default Dashboard;
