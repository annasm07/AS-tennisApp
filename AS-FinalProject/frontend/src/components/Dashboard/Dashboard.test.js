import {render, fireEvent} from '@testing-library/react-native';
import React from 'react';
import ConfigureStore from 'redux-mock-store';
import * as actions from '../../redux/actions/actionCreators';
import {Provider} from 'react-redux';
import Dashboard from './Dashboard';

jest.mock('../../redux/actions/actionCreators');

describe('When invoked a Dashboard component', () => {
  const mockStore = ConfigureStore();
  const store = mockStore({
    player: {
      record: [10, 3],
      name: 'Anna Sala',
      playedMatches: [
        {
          result: [
            {
              games: [0, 0, 0],
              name: 'Anna',
            },
            {
              games: [0, 0, 0],
              name: 'Sala',
            },
          ],
        },
      ],
    },
    tokens: ['abc', 'cde'],
    user: {user: {playerId: '123cde'}},
  });
  let navigation;
  beforeEach(() => {
    jest.spyOn(actions, 'getPlayerInfo').mockReturnValueOnce({type: ''});

    navigation = {
      navigate: jest.fn(),
    };
  });
  test('Should render dashboard component', () => {
    const dashboard = render(
      <Provider store={store}>
        <Dashboard navigation={navigation} />
      </Provider>,
    );
    expect(dashboard).toMatchSnapshot();
  });

  test('When Match is pressed, it shour navigate to stats', () => {
    const {getByTestId} = render(
      <Provider store={store}>
        <Dashboard navigation={navigation} />
      </Provider>,
    );
    const matchBox = getByTestId('matchBox');
    fireEvent.press(matchBox);
    expect(navigation.navigate).toHaveBeenCalled();
  });
});
