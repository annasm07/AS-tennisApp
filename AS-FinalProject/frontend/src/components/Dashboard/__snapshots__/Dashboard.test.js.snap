// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When invoked a Dashboard component Should render dashboard component 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "backgroundColor": "#E5E5E5",
            "borderRadius": 10,
            "flexDirection": "row",
            "margin": 15,
            "padding": 10,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "undefined",
            }
          }
          style={
            Object {
              "borderRadius": 100,
              "height": 80,
              "width": 80,
            }
          }
        />
        <View
          style={
            Object {
              "display": "flex",
              "flexDirection": "column",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 20,
                "marginTop": 10,
                "paddingLeft": 20,
                "textTransform": "capitalize",
              }
            }
          >
            Anna Sala
          </Text>
          <View>
            <View
              style={
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "pink",
                  "display": "flex",
                  "flexDirection": "row",
                  "height": 14,
                  "marginLeft": 20,
                  "marginTop": 15,
                  "width": 220,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#96F87E",
                      "height": 14,
                      "width": "50%",
                    },
                    Object {
                      "width": "76.92307692307693%",
                    },
                  ]
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#F68A5B",
                      "height": 14,
                      "width": "50%",
                    },
                    Object {
                      "width": "23.076923076923077%",
                    },
                  ]
                }
              />
            </View>
            <View
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginLeft": 20,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#7A7A7A",
                  }
                }
              >
                Wins / Losses
              </Text>
              <Text
                style={
                  Object {
                    "color": "#7A7A7A",
                  }
                }
              >
                 
                10
                 / 
                3
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="matchBox"
      >
        <View>
          <View
            style={
              Object {
                "backgroundColor": "#E5E5E5",
                "borderRadius": 10,
                "margin": 15,
                "marginBottom": 5,
                "padding": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "borderBottomColor": "#7A7A7A",
                  "borderBottomWidth": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingVertical": 8,
                }
              }
            >
              <Text>
                Anna
              </Text>
              <Text
                style={
                  Object {
                    "fontSize": 20,
                    "letterSpacing": 20,
                  }
                }
              >
                0
                0
                0
              </Text>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingVertical": 8,
                }
              }
            >
              <Text>
                Sala
              </Text>
              <Text
                style={
                  Object {
                    "fontSize": 20,
                    "letterSpacing": 20,
                  }
                }
              >
                0
                0
                0
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "backgroundColor": "#E5E5E5",
                "borderRadius": 10,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginBottom": 15,
                "marginHorizontal": 15,
                "padding": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#7A7A7A",
                }
              }
            >
              Invalid Date
            </Text>
            <Text
              style={
                Object {
                  "color": "#7A7A7A",
                }
              }
            >
              Stats &gt;
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
